test/test_index_string.cpp

```diff
@@ -1969,4 +1969,30 @@ TEST_TYPES(StringIndex_Insensitive_VeryLongStrings, non_nullable, nullable)
 }
 
 
+// Bug with case insensitive search on numbers that gives duplicate results
```

Comment:
It seems like these duplicate results were not being exposed via the query API. Am I mistaken about that, or was something in the query machinery skipping duplicates?